tools/gulptasks/dist-release.js

```diff
@@ -311,37 +355,69 @@ function checkIfLoggedInOnNpm() {
     }
 }
 
+/**
+ * Based on the product, checks if the code folder exists if it is not, compiles it.
+ */
+async function checkIfCodeExists() {
+    const codePaths = {
+        Highcharts: 'code/highcharts.js',
+        Grid: 'code/grid/grid-lite.js'
+    };
+
+    const filePath = codePaths[productName];
+    if (filePath && !fs.existsSync(filePath)) {
+        throw new Error(`${productName} code not found. Please run 'npm run dist' first.`);
+    }
+}
+/**
+ * Gets the dist repo path based on the product.
+ *
+ * @return {string} The dist repo path.
+ */
+function getDistRepo() {
+    return `../${productName.toLowerCase()}-dist/`;
+}
+
 /**
  * Performs all tasks for copying distribution contents, tagging (highcharts-dist) and releasing on npm.
  *
  * @return {Promise<*> | Promise | Promise} Promise to keep
  */
 async function release() {
+    // Set some variables
+    productName = argv.product || 'Highcharts';
+    releaseRepo = `${productName.toLowerCase()}-dist`;
+    pathToDistRepo = getDistRepo();
+
+    // Release
     const products = await getProductsJs();
-    const version = products[PRODUCT_NAME].nr;
+    const version = argv.release || products[productName].nr;
```

Comment:
We don't need to use the argv.release here, because its saved in the productsJs with the `npx gulp dist`